client/packages/system/src/Stock/api/hooks/useInventoryAdjustment.ts

```diff
@@ -0,0 +1,85 @@
+import { useCallback, useState } from 'react';
+import {
+  AdjustmentDirectionInput,
+  useAuthContext,
+  useGql,
+  useMutation,
+  useQueryClient,
+} from '@openmsupply-client/common';
+import {
+  InventoryAdjustmentReasonRowFragment,
+  StockLineRowFragment,
+} from '../../..';
+import { getSdk } from '..';
+
+type DraftInventoryAdjustment = {
+  direction: AdjustmentDirectionInput | null;
+  reason: InventoryAdjustmentReasonRowFragment | null;
+  adjustBy: number;
+  newNumberOfPacks: number;
+};
+
+export function useInventoryAdjustment(stockLine: StockLineRowFragment) {
+  // would usually query data here
+
+  // can then manage state (buffered too?) - can expose debounced update from here
+  const [draft, setDraft] = useState<DraftInventoryAdjustment>({
+    direction: null,
+    reason: null,
+    adjustBy: 0,
+    newNumberOfPacks: 0,
```

Comment:
Keen for thoughts on having this initial state be 0...

When in initial state, direction is null, so newNumberOfPacks input is disabled. 

When direction is set, newNumberOfPacks is set to stockLine.totalNumberOfPacks - this is so that new total is reset if you change the direction again.

In my ideal world, the default/initial state would be `stockLine.totalNumberOfPacks` even when the input is disabled. However, we agreed that when the user hits save, the form would reset. Not sure what the best way would be to reset the `newNumberOfPacks` field?

- Could optimistically set it to the `newNumberOfPacks` that you sent in the mutation?
- Otherwise would have to use some kind of buffered state situation after the stockLine is refetched and we have the new `totalNumberOfPacks` value?

Reply:
Is this data shape dictated by the mutation requirements? Because why do we need separate "direction" and "ajustBy" values? Can't we just "adjustBy" a negative value for reductions? Also, I'm not entirely sure why `newNumberOfPacks` doesn't just get reset/initialised with the current number -- perhaps I'm missing something?

Reply:
hmm, now that newNumberOfPacks doesn't have to be editable, it can be deduced based on adjustment and total number of packs right ? So doesn't have to be part of the state

If it was editable, maybe we can treat undefined = totalNumberOfPacks and like in the input `value={draft.newNumberOfPacks ?? totalNumberOfPcaks}`